src/libfetchers/path.cc

```diff
@@ -140,6 +141,20 @@ struct PathInputScheme : InputScheme
                 mtime = dumpPathAndGetMtime(absPath.string(), sink, defaultPathFilter);
             });
             storePath = store->addToStoreFromDump(*src, "source");
+        } else {
+            // To avoid copying the path again to the /nix/store, we need to add a cache entry.
+            ContentAddressMethod method = ContentAddressMethod::Raw::NixArchive;
+            auto fp = getFingerprint(store, input);
+            if (fp) {
+              auto cacheKey = fetchers::Cache::Key{"fetchToStore", {
+                  {"name", "source"},
+                  {"fingerprint", *fp},
+                  {"method", std::string{method.render()}},
+                  {"path", "/"},
+              }};
+
+              fetchers::getCache()->upsert(cacheKey, *store, {}, *storePath);
```

Comment:
I checked if I could do the insertion one layer down in fetch-to-store, but it feels a bit like layer violation because this layer shouldn't know about path inputs.

This currently affects for example: https://github.com/NixOS/nixpkgs/pull/388090

Reply:
Alternative would be to have some method that can be checked and will bypass the cache somehow i.e. returning a store path.